components/PreviewModal.js

```diff
@@ -0,0 +1,187 @@
+import { useState, useEffect } from "react";
+import Link from "../components/Link";
+import { IconContext } from "react-icons";
+import { FaRegComments } from "react-icons/fa";
+import requestIp from "request-ip";
+import PageHead from "../components/PageHead";
+import SingleLayout from "../components/layouts/SingleLayout";
+import MultiLayout from "../components/layouts/MultiLayout";
+import singleUser from "../config/user.json";
+import UserProfile from "../components/user/UserProfile";
+import UserTabs from "../components/user/UserTabs";
+import UserLinks from "../components/user/UserLinks";
+import UserMilestones from "../components/user/UserMilestones";
+import UserTestimonials from "../components/user/UserTestimonials";
+import UserEvents from "../components/user/UserEvents";
+import Page from "../components/Page";
+
+export default function Preview({ isOpen, toggle }) {
+  const [userData, setUserData] = useState();
+  const BASE_URL = process.env.NEXT_PUBLIC_BASE_URL;
+
+  useEffect(() => {
+    if (isOpen) {
+      handlePreviewData();
+    }
+  }, [isOpen]);
+
+  const handlePreviewData = () => {
+    let userPreviewData = localStorage.getItem("PREVIEW_PROFILE_DATA");
+    let parsedData = JSON.parse(userPreviewData);
+    setUserData(parsedData);
+    console.log(parsedData);

```

Comment:
I think the debug code can be removed?

Reply:
Thanks @eddiejaoude for reviewing the code. Will make all these changes which you suggested